go/processor/floria/precompiled.go

```diff
@@ -48,3 +54,215 @@ func getPrecompiledContract(address tosca.Address, revision tosca.Revision) (get
+
+var (
+	setBalanceMethodID []byte
+	copyCodeMethodID   []byte
+	swapCodeMethodID   []byte
+	setStorageMethodID []byte
+	incNonceMethodID   []byte
+)
+
+func init() {
+	abi, err := abi.JSON(strings.NewReader(stateContractABI))
+	if err != nil {
+		panic(err)
+	}
+
+	for name, constID := range map[string]*[]byte{
+		"setBalance": &setBalanceMethodID,
+		"copyCode":   &copyCodeMethodID,
+		"swapCode":   &swapCodeMethodID,
+		"setStorage": &setStorageMethodID,
+		"incNonce":   &incNonceMethodID,
+	} {
+		method, exist := abi.Methods[name]
+		if !exist {
+			panic("unknown EvmWriter method")
+		}
+
+		*constID = make([]byte, len(method.ID))
+		copy(*constID, method.ID)
+	}
+}
+
+var ErrExecutionReverted = fmt.Errorf("execution reverted")
+var ErrOutOfGas = fmt.Errorf("out of gas")
+
+func handleStatePrecompiledContract(state tosca.WorldState, sender tosca.Address, address tosca.Address, input []byte, gas tosca.Gas) (tosca.CallResult, bool) {
+	if address != StateContractAddress {
+		return tosca.CallResult{}, false
+	}
+	gas, err := statePrecompiledContract(state, sender, input, gas)
+	return tosca.CallResult{
+		Success: err == nil,
+		Output:  nil,
+		GasLeft: tosca.Gas(gas),
+	}, true
+}
+
+func statePrecompiledContract(state tosca.WorldState, sender tosca.Address, input []byte, gas tosca.Gas) (tosca.Gas, error) {
```

Comment:
nit: `runStatePrecompiledContract`?

Also the other functions in this file have a 'state' prefix to -- I guess -- artificially group them and avoid name collisions. Two ideas to group them otherwise would be
- define an empty "StateContract" struct and make them members of this type
- define a package "state_contract" and move the implementation into this package

Wdyt?

Reply:
I merged both functions and renamed the resulting one to `handleStateContract`, I think this is clear now.